app/js/model/list_model.js

```diff
@@ -48,12 +48,36 @@ export default class ListModel extends Model {
     return detail;
   }
 
+  fillAppDetailsFromEntryPoint(app, entry_point) {
+    var detail = Object.create(null);
+    detail.manifestURL = app.manifestURL;
+    detail.description = app.manifest.description;
+    detail.author = app.manifest.developer ? app.manifest.developer.name : '';
+    detail.app = app;
+    detail.name = app.manifest.entry_points[entry_point].name;
+    detail.icon = IconHelper.getIconURL(app, app.manifest.entry_points[entry_point].icons);
+    detail.entry_point = entry_point;
+    return detail;
+  }
+
   getAppList(allApps) {
     return new Promise((resolve, reject) => {
       var installedApps = Object.create(null);
       var filterList = this.filterApps(allApps);
-      filterList.forEach(app =>
-        installedApps[app.manifestURL] = this.fillAppDetails(app));
+      filterList.forEach(app => {
+        // Check for Communications app, only app that uses
+        // multiple entry_points in manifest
+        if (app.manifestURL == 'app://communications.gaiamobile.org/manifest.webapp') {
```

Comment:
+1, app.manifest.entry_points works and remove one more hard coded variable! will update patch. Thanks!